NumericOutcome.md

```diff
@@ -0,0 +1,437 @@
+An array of integers corresponds to a single event equal to the concatenation of these integers (interpreted in base `B`).
+
+### Example
+
+Before specifying the algorithm, let us run through a general example.
+
+Consider the range `[(prefix)wxyz, (prefix)WXYZ]` where `prefix` is some string of digits in base `B` which
+`start` and `end` share and `w, x, y, and z` are the unique digits of `start` in base `B` while `W, X, Y, and Z`
+are the unique digits of `end` in base `B`.
+
+To cover all cases while looking at as few digits as possible in this (general) range we need only consider
+`(prefix)wxyz`, `(prefix)WXYZ` and the following cases:
+
+```
+wxy(z+1), wxy(z+2), ..., wxy(B-1),
+wx(y+1)_, wx(y+2)_, ..., wx(B-1)_,
+w(x+1)__, w(x+2)__, ..., w(B-1)__,
+
+(w+1)___, (w+2)___, ..., (W-1)___,
+
+W0__, W1__, ..., W(X-1)__,
+WX0_, WX1_, ..., WX(Y-1)_,
+WXY0, WXY1, ..., WXY(Z-1)
+```
+
+where `_` refers to an ignored digit (an omission from the array of integers) and all of these cases have the `prefix`.
+I refer to the first three rows as the **front groupings** the fourth row as the **middle grouping** and the last three rows
+as the **back groupings**.
+
+Notice that the patterns for the front and back groupings are nearly identical, and that in total the number of CETs that
+will be required to cover the range will be equal to the sum of the unique digits of `end` plus the sum of `B-1` minus the
+unique digits of `start`.
+This means that the number of CETs required to cover a range of length `L` will be `O(B*log_B(L))` because `log_B(L)`
+corresponds to the number of unique digits between the start and end of the range and for each unique digit a row is
+generated in both the front and back groupings of length at most `B-1 ` which corresponds to the coefficient in the order bound.
+This counting shows us that base 2 is the optimal base to be using in general cases as it will outperform all larger bases
+in both large and small ranges in general.
+
+Note that there are two more possible optimizations to be made, which I call the **row optimization**, using the outliers `wxyz` and `WXYZ`.
+If `z=0` then the entire first row can be replaced with `wxy_` and if `Z=B-1` then the entire last row can be replaced with `WXY_`.
+There are another two possible optimizations in the case where the front or back groupings are not needed, which
+I call **grouping optimization**, that again use the outliers to the above pattern `wxyz` and `WXYZ`.
+For the front this is the case when `x=y=z=0` so that the front groupings can be replaced with `w___`.
+Likewise if `X = Y = Z = B-1` then the back groupings can be replace with `W___`.
+Lastly, if both grouping optimizations can be made, `w = 0` and `W = B-1` then a **total optimization** can be made and the
+whole range can be represented using only a single CET corresponding to `(prefix)____`.
+
+### Algorithms
```

Comment:
The functions in these should probably be written in a more universal language like python or just psdueo code

Reply:
Curious to see what people who code in non-scala think about the readability, please leave comments here!

Reply:
I agree with @benthecarman. Tbh it's fine by me since I got a little bit used to read Scala by now, but it definitely adds to the complexity of something that is already not that easy. I think Python would probably be better. I'm not sure for pseudo code, personally I often find it harder to read than actual code. Also still good to link to your implementation.

Reply:
As C# guy I can understand the code.
But I think if Scala is chosen, at the very least it should not use fancy constructions.